### PR TITLE
feat: add wish.Command and wish.Cmd 

### LINKED ISSUE #228: wish-exec has issues starting a bash shell (outputs to server stdout | error `Inappropriate ioctl for device`)
# What

The wish-exec example works fine with vim but when trying to exec an interactive bash process the following occurs.

## Outputs to server Stdout

Relying on `bubbletea.ExecProcess` to set the default Std(in,out,err) for the *exec.Cmd results in the Stdout of the bash process being on the server's stdout.

https://github.com/charmbracelet/bubbletea/blob/v0.25.0/exec.go#L112C16-L112C22
https://github.com/charmbracelet/bubbletea/blob/v0.25.0/options.go#L35C23-L35C32
https://github.com/muesli/termenv/blob/v0.15.2/output.go#L184

## Output to client Stdout but bash reports an error at startup

Explicitly setting the Std(in,out,err) to the `pty.Slave`, as the makeOptions function does results in the bash process outputting to the client but displaying the following error upon startup.

https://github.com/charmbracelet/wish/blob/main/bubbletea/tea_unix.go#L22

```
bash: cannot set terminal process group (1632577): Inappropriate ioctl for device
bash: no job control in this shell
```

Ctrl+c isn't registered by shell either.

# Reproduction

The following is a diff with a runnable example of this error.

```diff
diff --git a/examples/wish-exec/main.go b/examples/wish-exec/main.go
index 1137779..1dbbdcd 100644
--- a/examples/wish-exec/main.go
+++ b/examples/wish-exec/main.go
@@ -93,6 +93,32 @@ func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 				return vimFinishedMsg{err: err}
 			})
 			return m, cmd
+		case "f":
+			log.Info("pressed f: launching bash (server stdout)")
+			// Output shows up on server's stdout, no error messages
+			c := exec.Command("bash", "-il")
+			cmd := tea.ExecProcess(c, func(err error) tea.Msg {
+				if err != nil {
+					log.Error("bash (server stdout) finished", "error", err)
+				}
+				return vimFinishedMsg{err: err}
+			})
+			return m, cmd
+		case "b":
+			log.Info("pressed b: launching bash (client stdout)")
+			pty, _, _ := m.sess.Pty()
+			// Output shows up on client stdout, but there is an error message
+			c := exec.Command("bash", "-il")
+			c.Stdin = pty.Slave
+			c.Stdout = pty.Slave
+			c.Stderr = pty.Slave
+			cmd := tea.ExecProcess(c, func(err error) tea.Msg {
+				if err != nil {
+					log.Error("bash (client stdout) finished", "error", err)
+				}
+				return vimFinishedMsg{err: err}
+			})
+			return m, cmd
 		case "q", "ctrl+c":
 			return m, tea.Quit
 		}
```
Peek 2024-01-19 20-18

### PR DESCRIPTION
**Pull request by caarlos0**
The wish-exec example with vim worked because neovim was not using
STDERR.

Bubbletea doesn't have a concept of stdout and stderr, just output, so
`tea.ExecProcess` sets the `exec.Cmd` stderr to `os.Stderr`.

This would fail for bash, for instance.

This also introduces a `wish.Cmd` type and a `wish.Command`
function to properly set up a `wish.Cmd` based on `ssh.Session` (and
optionally a Pty), which can then be used with `tea.Exec`.

Finally, it adds to the wish-exec example, including the `s` key to run
a shell (bash).

closes #228

### CODE REVIEW

File: cmd.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	c := exec.Command(name, args...)
+	pty, _, ok := s.Pty()
+	if !ok {
+		c.Stdin, c.Stdout, c.Stderr = s, s, s
+		return &Cmd{cmd: c}
+	}
+
+	return &Cmd{c, &pty}
+}
+
```
Comment (ghthor) [inline cmd.go:27]:
```suggestion
	return command(s, c)
}

func CommandContext(ctx context.Context, s ssh.Session, name string, args ...string) *Cmd {
	c := exec.CommandContext(ctx, name, args...)
	return command(s, c)
}

func command(s ssh.Session, c *exec.Cmd) *Cmd {
	pty, _, ok := s.Pty()
	if !ok {
		c.Stdin, c.Stdout, c.Stderr = s, s, s
		return &Cmd{cmd: c}
	}

	return &Cmd{c, &pty}
}

```

Add in a `CommandContext` wrapper as well.

File: cmd.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+// SetStdout conforms with tea.ExecCommand.
+func (*Cmd) SetStdout(io.Writer) {}
+
+// Run runs the program and waits for it to finish.
+func (c *Cmd) Run() error {
+	if c.pty == nil {
+		return c.cmd.Run()
+	}
+
+	if err := c.pty.Start(c.cmd); err != nil {
```
Comment (ghthor) [inline cmd.go:64]: Ah, this makes a lot of sense. Had I manually added these SysProcAttr IN ADDITION to manually setting the Std{in,out,err} equal to the PTY Slave, the bash startup would stop complaining on startup about disabling job control.

```go
        c := exec.Command("bash", "-im")
	c.Stdin, c.Stdout, c.Stderr = i.Slave, i.Slave, i.Slave
	if c.SysProcAttr == nil {
		c.SysProcAttr = &syscall.SysProcAttr{}
	}
	c.SysProcAttr.Setctty = true
	c.SysProcAttr.Setsid = true
```

https://github.com/charmbracelet/ssh/blob/8ca513fa538f/pty.go#L69
https://github.com/charmbracelet/ssh/blob/8ca513fa538f49c49f851b88a1e19a5141b52583/pty_unix.go#L68

File: cmd_test.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/charmbracelet/ssh"
+	"github.com/charmbracelet/wish/testsession"
+)
+
```
Comment (ghthor) [inline cmd_test.go:11]:
```suggestion

var _ tea.ExecCommand = &Cmd{}

```

Static assertion that this type does indeed implement the `tea.ExecCommand` interface.

File: cmd.go [hunk 4]
```diff
@@ -25,6 +27,16 @@ func Command(s ssh.Session, name string, args ...string) *Cmd {
...
+// Command sets stdin, stdout, and stderr to the current session's PTY slave.
+//
+// If the current session does not have a PTY, it sets them to the session
+// itself.
+//
+// This will call CommandContext using the session's Context.
+func Command(s ssh.Session, name string, args ...string) *Cmd {
+	return CommandContext(s.Context(), s, name, args...)
+}
+
```
Comment (ghthor) [inline cmd.go:39]: :+1: Even smarter

**Review by aymanbagabas — APPROVED** [review]

**Pull request merged**